src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -627,10 +629,14 @@ void CascadiaSettings::LayerJson(const Json::Value& json)
 void CascadiaSettings::_LayerOrCreateProfile(const Json::Value& profileJson)
 {
     // Layer the json on top of an existing profile, if we have one:
-    auto pProfile = _FindMatchingProfile(profileJson);
-    if (pProfile)
+    auto profileIndex{ _FindMatchingProfileIndex(profileJson) };
+    if (profileIndex)
     {
-        pProfile->LayerJson(profileJson);
+        // add a new inheritance layer, and apply json values to child
+        auto parentProj{ _profiles.GetAt(*profileIndex) };
+        auto parent{ winrt::get_self<Profile>(parentProj) };
+        auto childImpl{ parent->CreateChild() };
+        childImpl->LayerJson(profileJson);

```

Comment:
don't you need to.. like, overwrite the parent in the array with the child now that it's inherited?

Reply:
haha. I did all the work of getting the index, then just didn't use it